server/src/main/java/org/opensearch/cluster/routing/IndexRoutingTable.java

```diff
@@ -466,21 +466,20 @@ public Builder initializeAsRemoteStoreRestore(
             }
             for (int shardNumber = 0; shardNumber < indexMetadata.getNumberOfShards(); shardNumber++) {
                 ShardId shardId = new ShardId(index, shardNumber);
-                if (indexShardRoutingTableMap.containsKey(shardId) == false) {
+                if (forceRecoveryPrimary == false && indexShardRoutingTableMap.containsKey(shardId) == false) {
                     throw new IllegalStateException("IndexShardRoutingTable is not present for shardId: " + shardId);
                 }
                 IndexShardRoutingTable.Builder indexShardRoutingBuilder = new IndexShardRoutingTable.Builder(shardId);
                 IndexShardRoutingTable indexShardRoutingTable = indexShardRoutingTableMap.get(shardId);
-                if (restoreAllShards || indexShardRoutingTable.primaryShard().unassigned()) {
+                if (forceRecoveryPrimary || indexShardRoutingTable == null || indexShardRoutingTable.primaryShard().unassigned()) {
```

Comment:
why would indexShardRoutingTable be null?

Reply:
if indexmetadata is not present, ie full cluster restore, routing table would not be present